### PR TITLE
Rebalances The Medibot, "Hold on I'm Coming (to give you surgery bonuses???)"

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
## About The Pull Request

Assume none of this is true for anything but the "normal" medibots you craft. Derelict and others function like old.

- Medibots now only heal brute/burn

- Medibots heal considerably less

- Medibots heal per tech has been lessened.

**BUT WAIT THERE'S MORE**

- Medibots now will "prepare you for surgery". This gives you a special trait which provides bonuses to a variety of surgical successes (and sometimes even lessens failures!).

- Leveling Medibots now increases the time you have this buff!

- If you do not have the buff, medibots will try to give it to you even if you aren't wounded!

Examples include TW healing 20% more per run, Stomach pump removing more reagents, Organ healing setting your damage to lower thresholds, and more!

Minorly, I have raised the abductor plastic surgery choice to 10 from 9.

## Why It's Good For The Game

I am not a fan of the crutch that is medibots but I also wanted it to be a tangible bonus to having medbay interact with Roboticists, so I (hopefully) solidified them as a role that would benefit medical staff rather than act as a replacement.

Removing the ox/tox was because it made little sense for those to be healed by the robot since there is not even an equivalent to these for surgery by doctors. These damages are by design chemistry oriented.

"who do I go see if I have minor damage then if not the bots?" chemistry, the roaming paramedics, doctors if you're terribly pressed, or that guy with the toolbox and bedsheets. You could theoretically still use the medibot but it's going to take more time now and you're losing value if you aren't taking advantage of the buff.

## Changelog
:cl: Cobby
balance: crafted medibots now "prepare you for surgery", providing you with additional benefits when undergoing surgery. Duration scales with tech like healing.
balance: crafted medibots heal less and only brute/burn.
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
Is there a way to see if someone's prepared for surgery just by examining them? Maybe a readout in surgery computers?

**Comment by Tlaltecuhtli** [comment]
can you make each color only heal that type and white a stabilizer (gets u out of crit and thats it)

**Comment by ExcessiveUseOfCobblestone** [comment]
I'd rather than not be seen as a healing tool at all, but rather an item that supports medical endeavours(as well as report out critical patients if found).

**Comment by Learner10** [comment]
I've seen some rather interesting comebacks from the brink thanks to a medibot finding the right person at the right time. I'm not against a rebalance, but I'm also happy that they're not being gutted health wise.

Though ultimately, I must admit that I'll miss those little guys helping you with toxin damage. No more medibots to treat my prisoner meth addiction.

**Comment by ATH1909** [comment]
damn, guess I'll just die to rads now because I can't use a pocket medibot to keep me alive

**Comment by ExcessiveUseOfCobblestone** [comment]
as you should?

File: code/modules/surgery/stomachpump.dm [hunk 1]
```diff
@@ -44,6 +44,8 @@
 				"<span class='notice'>[user] forces [H] to vomit, cleansing their stomach of some chemicals!</span>",
 				"[user] forces [H] to vomit!")
 		H.vomit(20, FALSE, TRUE, 1, TRUE, FALSE, purge = TRUE) //called with purge as true to lose more reagents
+		if(HAS_TRAIT(target,TRAIT_SURGERYPLUS))
+			target.reagents?.remove_all(5) //additional purging that is irretrievable (in case someone had the crazy idea to make reagents extractable via vomit).
```
Comment (ATH1909) [inline code/modules/surgery/stomachpump.dm:48]: can't you scoop up vomit pools off of the ground using a beaker?

File: code/modules/surgery/revival.dm [hunk 2]
```diff
@@ -57,6 +57,8 @@
 		"<span class='notice'>[user] prepares to shock [target]'s brain with [tool].</span>",
 		"<span class='notice'>[user] prepares to shock [target]'s brain with [tool].</span>")
 	target.notify_ghost_cloning("Someone is trying to zap your brain. Re-enter your corpse if you want to be revived!", source = target)
+	if(HAS_TRAIT(target,TRAIT_SURGERYPLUS))
```
Comment (ATH1909) [inline code/modules/surgery/revival.dm:60]: this will very, very rarely occur, as medibots don't target dead people, right?

**Comment by zxaber** [comment]
If you're open to alternative rebalances, an idea I've seen is to have medibots use an expendable resource, such as a scalple that needs to be replaced after so many uses. That would still allow medibots to heal like they could before, and be useful in the halls, but not really a doctor replacement unless Robo is constantly replacing scalples in them.

**Comment by Archanial** [comment]
Make them heal you for x hp but only once instead of healing you to the full.

**Comment by ThePainkiller** [comment]
Cobby can you explain why are you still using _"successes"_ and _"failures"_ since success chance was removed on october 2019 and ghetto tools only have a penalty to speed? The only way you can fail now is _on purpose._

Also, I don't think this is such a **BAD** nerf, considering why would a medibot tend mixed wounds as equally fast as a Medical Doctor with Advanced Tools and Legendary skill? Why am I even training to such heights if a mere bot can perform equally as good? Do we relegate the wound tending to the bot and then make everything else?  The change does makes sense considering they're just bots, they **SHOULDN'T** be **_THAT_** good. But assisting on surgery? That's actually helpful and a nice and welcome change.
It's sad they won't heal toxins or oxyloss, considering back in the day they used to actually inject chems to heal your damage types (wich meant death if you had two or more medibots healing you at the same time), I wish the medibots at least injected a bit of epinephrine to keep patients stabilizing since you want them to be "surgery assistants" rather than literal roaming medics

**Comment by trexter555** [comment]
stealth slimeperson buff

**Comment by wesoda25** [comment]
Oh boy more medical rebalances

**Comment by Shadowflame909** [comment]
if they heal considerably less

how does this effect the emagged/given poisoned reagents medibot?

**Comment by ATH1909** [comment]
So, uh, just to be clear, I don't like this PR.

For one thing, press F for pAI medibots (and non-medical borgs using medibots as a way to save dying humans).

Also, Medbay is gonna burn through oxy and tox meds (ESPECIALLY for recent defibbees or irradiated people, respectively) WAY faster now. 

Also also, I feel like medibots have been one of the things that have been making playing medical doctor more bearable. Tend wounds takes half a century to do (even on a stripped target) and is boring as fuck (click on your patient 3 times, then alt+tab), and medibots allowed you to automate that process. I also feel like medical doctors are still gonna strap patients to stasis beds next to medibots to heal them, but now it'll just take 2.5x longer for those patients to be able to leave Medbay.

Also also also, I really need to reiterate that being irradiated is going to fuck your round over even more than it did before. If you don't have access to a hardsuit storage device, hacking tools, and a power source, you can kiss your round goodbye as soon as you get severely irradiated (no matter how many anti-rad meds you chug). Pocket medibots at least allowed you to return to work in isolated jobs, but now, with medibots unable to heal tox damage... I don't think that they'll be enough.

**Comment by ATH1909** [comment]
The surgery bonuses here are kind of pathetic as well. Woohoo, 10 less organ damage on an organ that's going to heal back to full health naturally anyway!

**Comment by ATH1909** [comment]
tldr;

Medibots currently serve as NPCs who'll cover the most boring parts of Medbay (healing up people who just have some booboos from the primary 4 damage types). Nerfing them into uselessness is going to make Medbay work more tedious for the doctors, and nerfing them into sort of usefulness is going to cause the meta to not change (just take longer to heal patients), so Medbay will become more tedious for the patients.

Am I making any sense, or am I just rambling at 5 AM again?

**Comment by ThePainkiller** [comment]
> if they heal considerably less
> 
> how does this effect the emagged/given poisoned reagents medibot?

Read the code, they inject less chloral.

**Comment by ATH1909** [comment]
> > if they heal considerably less
> > how does this effect the emagged/given poisoned reagents medibot?
> 
> Read the code, they inject less chloral.

that's an F for revenants too, then

**Comment by ATH1909** [comment]
do unupgraded emagged medibots even inject chems at a rate that exceeds the rate that the liver will try to purge them at?

**Comment by ThePainkiller** [comment]
> Am I making any sense, or am I just rambling at 5 AM again?

As a current med player who has to deal with the daily stress of an always understaffed medbay (unless there's three specific people I can count on being competent) you make a lot of sense and valid points. I do admit however that medibots being cheap and easy to make while doing the _"boring"_ wound tending afk aspect of making someone go vertical again feels a bit overpowered, considering they are doing it at a speed that equals a Legendary surgeon with Advanced Tools if they are at 400% efficiency (Experimental Surgeries node researched). This change is to eliminate that "cheese" that coders hate so much because people are clever enough to find workarounds to the tedious methods we are forced to undertake for the sake of making a person return from spectator mode, or the powergamers that set a medibot to not wander and drag it around to have it patch them up on the fly ~~(like Kevinz000 did on that one wizard round years ago on Sibil)~~

> The surgery bonuses here are kind of pathetic as well. Woohoo, 10 less organ damage on an organ that's going to heal back to full health naturally anyway!

The 10 hp on organ healing is indeed neglectable considering that organdamage-fixing surgeries are coded in such a way they are meant to restore organs to at least basic functionality. If they offered a 10% speed boost for surgeries that'd make it an actual useful alternative.

**Comment by XDTM** [comment]
Giving a boost to surgeries might be something nice to give, but the point of bots is that they can move around, seek their targets and do their thing. If their best use scenario is to park them in front of surgery, their function might as well be placed on some static piece of machinery instead.

File: code/modules/surgery/stomachpump.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/stomachpump.dm:48]: I cannot remember off the top of my head, but good thing I added this then lol

File: code/modules/surgery/revival.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/revival.dm:60]: correct

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 3]
```diff
@@ -39,7 +39,11 @@
 /// Don't spam the "HEY I'M COMING" messages
 	var/last_newpatient_speak = 0
 /// How much healing do we do at a time?
-	var/heal_amount = 2.5
+	var/heal_amount = 1
+/// How long do we apply the surgery bonus trait?
+	var/surgery_bonus_trait_duration = (15 SECONDS)
```
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/bot/medbot.dm:44]: white kits are more common, they aren't competitive.

Blu/Gre kits do nothing special unless the bot does not apply the trait, in which case it would function like old.

**Comment by Kaffe-work** [comment]
please do not nerf medbay any further, the less time I have to spend with cuck docs the better.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Cobby can you explain why are you still using "successes" and "failures" since success chance was removed on october 2019 and ghetto tools only have a penalty to speed? The only way you can fail now is on purpose.

They still exist, it's just that failure chance becomes translated into duration first followed by actual failure chance.

I'll do the medkit color healing instead so they are hyperfocused robots similar to beepsky (stuns but does not bring to brig), floortile bot, clean bot (doesnt remove trash just scrubs).

> Medibots currently serve as NPCs who'll cover the most boring parts of Medbay (healing up people who just have some booboos from the primary 4 damage types) for you, bolstering the ranks of an ofter-understaffed department. Nerfing them into uselessness would just make Medbay work more tedious for the doctors, and nerfing them into sort of usefulness wouldn't cause the meta to change (and instead just make it take longer for said meta-strat to heal patients), so getting healed in Medbay will become more tedious for the patients who have to wait there longer.

make surgery fun. This is assuming you're doing TW on living individuals, in which case I have to ponder why unless you just wanna show off your farmed surgery skill or something. Chemistry is a much better option and maps should support MDs getting into the apothecary.

**Comment by Learner10** [comment]
> I'll do the medkit color healing instead so they are hyperfocused robots similar to beepsky (stuns but does not bring to brig), floortile bot, **clean bot (doesnt remove trash just scrubs)**.

Clean bots do actually remove trash. They'll spray things with acid. They just have a small list of what actually counts for trash (if it counts towards the cargo trash bounty, it gets the acid bath).

> make surgery fun. This is assuming you're doing TW on living individuals, in which case I have to ponder why unless you just wanna show off your farmed surgery skill or something. Chemistry is a much better option and maps should support MDs getting into the apothecary.

Agreed. I usually just carry a pair of medical spray cans with basic C2's. The people who can walk in can usually be treated with a spray, and a couple of sutures and meshes.

**Comment by ExcessiveUseOfCobblestone** [comment]
cant be right all the time 😏

**Comment by ATH1909** [comment]
Chemistry should be for more portable meds; stuff you pick up from Chemistry and carry on you for most of the round so that you can heal yourself and others on the fly, not stuff that Medbay needs in order to function.

**Comment by ThePainkiller** [comment]
> Chemistry should be for more portable meds; stuff you pick up from Chemistry and carry on you for most of the round so that you can heal yourself and others on the fly, not stuff that Medbay needs in order to function.

Cobby, pushing changes that force people to rely on the chems they coded? No way!

**Comment by ExcessiveUseOfCobblestone** [comment]
The chem needs to be viable enough to work as a practically standalone option for healing but not viable enough to be used along with something that also heals (IE surgery) for max effectiveness?

Pretending that they dont have access to a very powerful item like the dispenser, which overlaps considerably with their current job, makes no sense.

**Comment by Rohesie** [comment]
> 
> 
> > Chemistry should be for more portable meds; stuff you pick up from Chemistry and carry on you for most of the round so that you can heal yourself and others on the fly, not stuff that Medbay needs in order to function.
> 
> Cobby, pushing changes that force people to rely on the chems they coded? No way!

Let's avoid this kind of passive-aggressive conspiratorial comments, please.
Focus on giving feedback on gameplay and whatnot, not on attacking the coders like that.
